### PR TITLE
Show case data streams from ui

### PR DESCRIPTION
**Pull request by serj-lotutovici**

### CODE REVIEW

File: app/src/main/java/com/serjltt/devfest/weather/mvp/Presenter.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.serjltt.devfest.weather.mvp;
+
+import rx.Subscription;
+
+/** Base contract for a Presenter in a MVP implementation. */
+public interface Presenter<T> {
+  /** Binds a view an returns a subscription. */
```
Comment (vanniktech) [inline app/src/main/java/com/serjltt/devfest/weather/mvp/Presenter.java]: nit: and

File: app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
-  final Scheduler observeScheduler;
+public final class ForecastPresenter implements Presenter<ForecastMvp.View> {
+  private final RxUseCase<List<ForecastMvp.Model>> getForecastUseCase;
+  private final Scheduler subscribeScheduler;
+  private final Scheduler observeScheduler;
 
-  public ForecastPresenter(UseCase<List<ForecastMvp.Model>> getForecastUseCase,
-      Scheduler subscribeScheduler, Scheduler observeScheduler) {
+  @Inject ForecastPresenter(RxUseCase<List<ForecastMvp.Model>> getForecastUseCase,
+      @Named(RxModule.IO_SCHEDULER) Scheduler subscribeScheduler,
```
Comment (vanniktech) [inline app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java]: I'd rename it to ioScheduler personally

File: app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
+      @Named(RxModule.MAIN_SCHEDULER) Scheduler observeScheduler) {
```
Comment (vanniktech) [inline app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java]: Also here rename to mainScheduler personally

**Review by vanniktech — APPROVED** [review]

File: app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java [hunk 4]
```diff
@@ -1,38 +1,49 @@
...
+        .subscribe(city -> {
+          subscription.add(getForecastUseCase.stream(city)
+              .subscribeOn(ioScheduler)
+              .observeOn(mainThreadScheduler)
+              .doOnNext(name -> view.hideLoading())
+              .subscribe(view::showForecast, throwable -> {
+                view.hideLoading();
+                view.showError(throwable.getMessage());
+              }));
+        }));
```
Comment (serj-lotutovici) [inline app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java:46]: @vanniktech using flatMap unsubscribes the whole chain... if you have a better idea, I'm open for suggestions.

Comment (vanniktech) [reply app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java:46]: I don't quite get why it 'll unsubscribe flatMap should work here

Comment (serj-lotutovici) [reply app/src/main/java/com/serjltt/devfest/weather/show/forecast/presenter/ForecastPresenter.java:46]: It works, but only once. When `getForecastUseCase.stream(city)`complets the unsubscribe is propagate it to  `cityName` observable

**Pull request merged**